compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1648,6 +1648,11 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             .chain(
                 tcx.hir()
                     .body_owners()
+                    // Incremental performance optimization: only load typeck results for things that actually have inline consts
+                    .filter(|&def_id| {
+                        tcx.hir_owner_nodes(tcx.hir().body_owned_by(def_id).hir_id.owner)
```

Comment:
This is not granular at the body level, but at the owner level. So nested bodies that do their own typeck will check the whole owner's flag, instead of a flag just for its body.

I can resolve this before landing the PR, but it's only a perf optimization